homeassistant/components/litejet/__init__.py

```diff
@@ -1,49 +1,71 @@
 """Support for the LiteJet lighting system."""
-from pylitejet import LiteJet
+import asyncio
+import logging
+
+import pylitejet
 import voluptuous as vol
 
+from homeassistant.config_entries import SOURCE_IMPORT, ConfigEntry
 from homeassistant.const import CONF_PORT
-from homeassistant.helpers import discovery
+from homeassistant.core import HomeAssistant
 import homeassistant.helpers.config_validation as cv
 
-CONF_EXCLUDE_NAMES = "exclude_names"
-CONF_INCLUDE_SWITCHES = "include_switches"
+from .const import DOMAIN, PLATFORMS
 
-DOMAIN = "litejet"
+_LOGGER = logging.getLogger(__name__)
 
 CONFIG_SCHEMA = vol.Schema(
-    {
-        DOMAIN: vol.Schema(
-            {
-                vol.Required(CONF_PORT): cv.string,
-                vol.Optional(CONF_EXCLUDE_NAMES): vol.All(cv.ensure_list, [cv.string]),
-                vol.Optional(CONF_INCLUDE_SWITCHES, default=False): cv.boolean,
-            }
-        )
-    },
+    {DOMAIN: vol.Schema({vol.Required(CONF_PORT): cv.string})},
     extra=vol.ALLOW_EXTRA,
 )
 
 
 def setup(hass, config):
     """Set up the LiteJet component."""
+    if not hass.config_entries.async_entries(DOMAIN) and DOMAIN in config:
+        # No config entry exists and configuration.yaml config exists, trigger the import flow.
+        hass.async_create_task(
+            hass.config_entries.flow.async_init(
+                DOMAIN, context={"source": SOURCE_IMPORT}, data=config[DOMAIN]
+            )
+        )
+    return True
+
 
-    url = config[DOMAIN].get(CONF_PORT)
+async def async_setup_entry(hass: HomeAssistant, entry: ConfigEntry) -> bool:
+    """Set up LiteJet via a config entry."""
+    port = entry.data[CONF_PORT]
 
-    hass.data["litejet_system"] = LiteJet(url)
-    hass.data["litejet_config"] = config[DOMAIN]
+    try:
+        system = pylitejet.LiteJet(port)
+    except Exception:  # pylint: disable=broad-except
+        _LOGGER.error("Error connecting to the LiteJet MCP at %s", port)
+        return False
 
-    discovery.load_platform(hass, "light", DOMAIN, {}, config)
-    if config[DOMAIN].get(CONF_INCLUDE_SWITCHES):
-        discovery.load_platform(hass, "switch", DOMAIN, {}, config)
-    discovery.load_platform(hass, "scene", DOMAIN, {}, config)
+    hass.data[DOMAIN] = system
```

Comment:
```suggestion
    hass.data[DOMAIN][entry.entry_id] = system
```

Store it under `hass.data[DOMAIN][entry.entry_id]` in case they have multiple.

Reply:
Having multiple configured won't play nicely with the trigger implementation. For this reason configuring multiple is forbidden in the config flow.